[Timothy Ng] iP

timiditi opened the pull request:
# Duke *ver. T*
> "Wow, this software is great!" - Me, in some other context.

This is Tim's *incredible* version of the infamous Duke software. 

It has:
* a **lot** of work and green tea :tea: put into it
* potential
* features I guess

What I did to get into this mess:
1. Sign up for CS2103T
2. Not realise what I was getting into
3. Profit..?

Here's a tiny preview of some code you'll find in this project!
``` java
public
```
Pretty cool, if I do say so myself. (You might also spot `return` somewhere in the code too)

### To-do:
- [x] start this project
- [ ] finish this project

[Enjoy!](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

Commit:
Add Gradle support

timiditi committed:
Implement skeletal version of Duke

Echo commands by user and exit when user types 'bye'.

timiditi committed:
Implement add and list functionality

User input is now added to an item list. 'list' command prints the list.

timiditi committed:
Add 'Mark as Done' functionality

Task class created. User can mark task in list as done.

timiditi committed:
Add Todo, Event and Deadline Tasks

Implements TodoTask, EventTask, DeadlineTask classes as subclasses of Task.

Keying in  'done' without anything after no longer causes an error.

timiditi committed:
Handle wrong inputs using exceptions

Some wrong user inputs are now handled with try-catch blocks instead of if-else blocks.

Some redundant code removed.

runtest.sh can be used to semi-automate testing.

timiditi committed:
Add Delete function

Added delete function to remove tasks from list.

The number of tasks in list will now be printed whenever a new task is added or removed.

timiditi committed:
Add Save/Load function

Added save/load feature for itemList. Adjusted some code formatting to match module standards.

timiditi committed:
Add date parsing functionality

Event and Deadline tasks now parse dates input in yyyy-mm-dd format.

timiditi committed:
Merge pull request #1 from timiditi/branch-level-7

Add Save/Load function

timiditi committed:
Merge branch 'master' into branch-level-8

timiditi committed:
Merge pull request #2 from timiditi/branch-level-8

Add date parsing functionality

timiditi committed:
Fix date for EventTask

fixes bug in EventTask from manual merging of branches.

timiditi committed:
Refactor code to include more OOP elements

Added Ui, Parser, Storage and TaskList classes.
Refactored Duke class.
Added Command class and subclasses.

timiditi committed:
Place classes into packages

Created and moved classes into duke, duke.task and duke.command packages.

timiditi committed:
Add JUnit tests

added JUnit support and 3 JUnit test methods.

timiditi committed:
Add JavaDoc documentation to several methods

timiditi committed:
Update code to match module standard

Renamed 'exit' boolean in Duke to 'isExit'.

timiditi committed:
Add search function

Added ability to find tasks using the 'find' command.

timiditi committed:
Merge pull request #3 from timiditi/branch-A-JavaDoc

Add JavaDoc documentation to several methods

timiditi committed:
Merge pull request #4 from timiditi/branch-A-CodingStandard

Update code to match module standard

timiditi committed:
Fix import statement issues in Parser

timiditi committed:
Merge pull request #5 from timiditi/branch-level-9

Add search function

ziqing26 commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    @Test
+    public void delete_when_taskList_empty() {
```
Perhaps a more intuitive function name here? Something like `featureUnderTest_testScenario_expectedBehavior()` ?

ziqing26 commented on task-list.txt:
```diff
@@ -0,0 +1,5 @@
+T|0|xxx
```
Perhaps you do not need to commit your local storage to GitHub?

itsyme replied:
Perhaps your local storage might not need to be committed to GitHub

ziqing26 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,68 @@
+
+    /**
+     * Constructor for a Duke instance.
+     * @param filePath file path for file to save/load save data.
```
Please add an Empty line between description and parameter section for all JavaDocs. 🥲

itsyme replied:
Perhaps an empty line would make the code cleaner

ziqing26 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,76 @@
+            String taskType = line[0];
+            boolean isDone = Integer.parseInt(line[1]) != 0;
+            String taskName = line[2];
+            switch (taskType) {
```
The indentation for `case` inside `switch` should be of the same level as `switch`, i.e. please remove one level of indentation for `case` clauses:).

itsyme replied:
Perhaps the indentation for `case` clauses could be changed to follow the coding standard

yewmeng28 replied:
It would be better if `case` were the same indentation as the `switch`, as per the coding standard.

ziqing26 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,48 @@
+import duke.task.Task;
+import duke.task.TaskList;
+
+import java.io.IOException;
```
In your `src/main/java/duke/command/AddTaskCommand.java`, this import line is above other imports from duke, but here it is below the same set of imports. Perhaps make the import sequence the same?

itsyme replied:
Perhaps the imports could be in a uniform sequence across all files to make the code cleaner

ziqing26 commented on src/main/java/duke/task/EventTask.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " + this.date.format(DateTimeFormatter.ofPattern("d MMM yyyy")) + ")";
```
Perhaps you could break this line into two lines as it is a bit long. Or maybe you could introduce a local variable for part of the string.

ziqing26 reviewed (commented):
Overall I enjoy reading your code 😄  Just a few small nitpicks on coding standards.

ryanongra commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,202 @@
+     */
+    private Command addEvent(String[] input) {
+
+        String TIME_MARKER = " /at ";
```
Since you've capitalised it, might be better to declare it as a constant?

ryanongra commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,202 @@
+     */
+    private Command addDeadline(String[] input) {
+
+        String DEADLINE_MARKER = " /by ";
```
Likewise for DEADLINE_MARKER

itsyme commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,68 @@
+    private TaskList taskList;
+    private Ui ui;
+
+    private boolean isExit;
```
I like the naming of isExit

ryanongra commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,56 @@
+ */
+public class FindCommand extends Command {
+
+    private String[] searchInput;
```
Perhaps can rename the variable to reflect that it's an array, maybe searchKeywords or sth like that?

ryanongra commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,75 @@
+
+public class TaskList {
+
+    private ArrayList<Task> list;
```
Maybe a better name for this might be tasks? Cos currently it isn't really intuitive what the list will contain.

yewmeng28 replied:
Agreed that `tasks` will be a better name for the `ArrayList<Task>`.

ryanongra reviewed (commented):
LGTM! Just a few small things to fix :)

itsyme reviewed (commented):
Good job writing clean code! Keep up the good work!

yewmeng28 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,202 @@
+     */
+    private Command addTodo(String[] input) {
+        try {
+            String name = input[1];
```
Perhaps it would be better to name this variable `description` as it is more descriptive of what it is representing.

yewmeng28 reviewed (commented):
Generally looks good, with a few nitpicky minor tweaks to make the code look nicer! Keep up the good work!

timiditi committed:
Merge pull request #6 from timiditi/add-gradle-support

Add Gradle support

timiditi committed:
Add GUI to Duke

Added basic GUI to Duke, removed text-based UI.

timiditi committed:
Merge pull request #8 from timiditi/branch-level-10

Add GUI to Duke

timiditi committed:
Add custom exceptions and make dialog box resizeable

Many exceptions are similar.
Let's create custom DukeExceptions so they can be grouped together.

Dialog boxes are a fixed size.
Dialog is cut off beyond a certain point.
Let's make them resizeable according to the message length.

timiditi committed:
Make exiting the program less abrupt

Using the 'bye' command would immediately close Duke without an exit message.
This can seem too abrupt.
Let's create an exit message after the 'bye' command and close Duke after the user acknowledges the message.

timiditi committed:
Add Assertions

Adds assertions for Parser class.

timiditi committed:
Improve code quality

timiditi committed:
Merge pull request #9 from timiditi/branch-A-CodeQuality

Improve code quality

timiditi committed:
Merge branch 'master' into branch-A-Assertions

timiditi committed:
Merge pull request #10 from timiditi/branch-A-Assertions

Add assertions

timiditi committed:
Update README.md

timiditi committed:
Update README.md

timiditi committed:
Update EventTask to check for time clashes

B-DetectAnomalies

EventTasks currently only hold dates as data. Let's add time as well. EventTask now accepts datetime in the format yyyy-MM-dd HH:mm.

Users may want to know if they are adding an event that clashes with their other events. Let's add functionality to inform them if there is an overlap in the timing of the event they are adding.

timiditi committed:
Improve GUI and fix event clashing issue

timiditi committed:
Change find command to show task index

When the find command returns a list of results, it returns them with the items numbered incrementally from 1. This is not helpful for users who want to know what the exact index of the item is on their task list.
Let's change it such that instead of renumbering the search results, it shows their exact index number from the task list.

timiditi committed:
Update user guide

timiditi committed:
Add welcome message to Duke
